ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleRoomDatabaseTest.java

```diff
@@ -95,20 +100,70 @@ public void verifyGetAllArticles() {
         assertTrue("No article was saved.", mSavedArticles.size() > 0);
     }
 
-
     @Test
     public void verifyArticleHasAttachment() {
         for (Article article : mSavedArticles) {
             assertTrue(String.format("Article %s has no attachment", article.mTitle),
-                       mAttachmentDao.getTestableAttachmentsByArticle(article.mId).size() > 0);
+                    mAttachmentDao.getTestableAttachmentsByArticle(article.mkey).size() > 0);
         }
     }
 
-
     @Test
     public void verifyManifestHasArticles() {
         assertTrue("Manifest 0 should have articles", mAssociationDao.getTestableArticlesByManifestID("0").size() > 0);
         assertTrue("Manifest 1 should have articles", mAssociationDao.getTestableArticlesByManifestID("1").size() > 0);
         assertTrue("Manifest 2 should have articles", mAssociationDao.getTestableArticlesByManifestID("2").size() > 0);
     }
+
+
+    @Test
+    public void verifyArticleParseLogic() {
+
+        try {
+            Uri url = Uri.parse("https://stage.cru.org/content/experience-fragments/questions_about_god/english.999.json");
```

Comment:
by depending on a web resource this test is very brittle. You should embed a raw json file and load that to perform your parsing tests. That way if the web content ever changes it doesn't break your test